pyswitcherv2/switcher.py

```diff
@@ -350,17 +355,29 @@ def parse_args():
 
     return args
 
+def run(args, try_num):
+    if try_num >= g_num_of_retries:
+        exit_with_error("ERROR: Reached max num of retries (%d), exiting..." % g_num_of_retries)
+
+    try:
+    	mode = args.mode
+        if mode == 'get_state': 
+            return get_state(args.credentials_file)
+        
+        if mode == 'parse_pcap_file':
+            parse_pcap_file(args.pcap_file)
+        elif mode == 'on' or mode == 'off':
+            control(mode == 'on', args.time_min, args.credentials_file)
+        else:
+            exit_with_error("ERROR: unexpected mode")
+
+        return 0
+    except Exception:
+    	sleep_sec = 3 * try_num
+        print("Retrying in %d seconds" % sleep_sec)
+        time.sleep(sleep_sec)
+        return run(mode, args, try_num + 1)
```

Comment:
unnecessary argument:
`return run(mode, args, try_num + 1)`
should be:
`return run(args, try_num + 1)`